### DISCUSSION TITLE
Running dataloader concurrently in CPU with the model (in GPU) using pytorch lightning

### DISCUSSION THREAD

**Discussion by jmoork** [Lightning-AI/pytorch-lightning#17619] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Is it possible to run the dataloader in CPU concurrently with the GPU running the model forward processing in pytorch lightning? This essentially allows the CPU dataloader to fetch the next batch of data while GPU is processing the current batch.

**Comment by cszhz** [comment]
+1
